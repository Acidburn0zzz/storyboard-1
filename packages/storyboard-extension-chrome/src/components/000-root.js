import { Provider } from 'react-redux';
import React from 'react';
import App from './005-app';

const Root = ({ store }) =>
  <Provider store={store}>
    <App />
  </Provider>;

Root.propTypes = {
  store: React.PropTypes.object.isRequired,
};

export default Root;
