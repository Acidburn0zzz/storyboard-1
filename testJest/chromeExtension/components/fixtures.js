import { addDefaults } from 'timm';

export const BASE_COLORS = {
  colorClientBg: 'aliceblue',
  colorClientBgIsDark: false,
  colorClientFg: 'black',
  colorServerBg: 'rgb(214, 236, 255)',
  colorServerBgIsDark: false,
  colorServerFg: 'black',
  colorUiBg: 'white',
  colorUiBgIsDark: false,
  colorUiFg: 'rgb(64, 64, 64)',
};

export const DARK_COLORS = {
  colorClientBg: 'rgb(17, 22, 54)',
  colorClientBgIsDark: true,
  colorClientFg: 'white',
  colorServerBg: 'rgb(14, 11, 33)',
  colorServerBgIsDark: true,
  colorServerFg: 'white',
  colorUiBg: 'black',
  colorUiBgIsDark: true,
  colorUiFg: 'white',
};

export const EMPTY_MAIN_STORY = {
  fExpanded: true,
  fHierarchical: true,
  fOpen: true,
  fWrapper: true,
  records: [
    {
      fExpanded: true,
      fHierarchical: true,
      fMain: true,
      fOpen: true,
      fServer: false,
      fStoryObject: true,
      id: 'main_0',
      lastAction: 'CREATED',
      numRecords: 0,
      pathStr: 'records/0',
      records: [],
      src: 'main',
      status: undefined,
      storyId: 'main_0',
      t: 0,
      title: 'Client',
    },
    {
      fExpanded: true,
      fHierarchical: true,
      fMain: true,
      fOpen: true,
      fServer: true,
      fStoryObject: true,
      id: 'main_1',
      lastAction: 'CREATED',
      numRecords: 0,
      pathStr: 'records/1',
      records: [],
      src: 'main',
      status: undefined,
      storyId: 'main_1',
      t: 0,
      title: 'Server',
    },
  ],
};

export const buildLogRecord = (options = {}) => addDefaults(options, {
  id: 'id1',
  hubId: 'hubId1',
  version: 3,
  fStory: false,
  fServer: false,
  storyId: 'story1',
  t: 0,
  src: 'main',
  level: 30,
  msg: 'Example message',
});

export const buildStory = (options = {}) => {
  const story = addDefaults(options, {
    id: 'id1',
    hubId: 'hubId1',
    version: 3,
    fStory: true,
    fServer: false,
    storyId: 'story1',
    t: 0,
    src: 'main',
    level: 30,
    fRoot: false,
    title: 'Story title',
    parents: [],
    // client-side
    fStoryObject: true,
    fOpen: false,
    fMain: false,
    status: undefined,
    fExpanded: true,
    fHierarchical: true,
    records: [],
  });
  story.records.unshift({
    id: `${story.id}.creation`,
    hubId: story.hubId,
    version: story.version,
    fStory: true,
    fServer: story.fServer,
    storyId: story.storyId,
    t: story.t,
    src: story.src,
    level: story.level,
    fRoot: story.fRoot,
    title: story.title,
    action: 'CREATED',
  });
  if (!story.fOpen) {
    story.records.push({
      id: `${story.id}.closure`,
      hubId: story.hubId,
      version: story.version,
      fStory: true,
      fServer: story.fServer,
      storyId: story.storyId,
      t: story.records[story.records.length - 1].t,
      src: story.src,
      level: story.level,
      fRoot: story.fRoot,
      title: story.title,
      action: 'CLOSED',
    });
  }
  story.numRecords = story.records.length;
  return story;
};
