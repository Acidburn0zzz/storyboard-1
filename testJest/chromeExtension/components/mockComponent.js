/* eslint-disable react/prop-types */

const React = require('react');

module.exports = (name) => (props) => (
  <div dataMockType={name} {...props}>
    <div style={{ fontWeight: 'bold', color: 'blue' }}>{name}</div>
    {props.children}
  </div>
);
