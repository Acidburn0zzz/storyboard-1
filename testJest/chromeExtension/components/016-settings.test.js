/* eslint-env jest */
/* eslint-disable global-require, import/newline-after-import */
import React from 'react';
import renderer from 'react-test-renderer';
import { Floats } from 'giu';
import { _Settings as Settings } from '../../../lib/chromeExtension/components/016-settings';
import { BASE_COLORS } from './fixtures';

jest.mock('react-dom');

describe('Settings', () => {
  it('renders correctly', () => {
    const settings = {
      colorClientBg: 'aliceblue',
      colorClientBgIsDark: false,
      colorClientFg: 'black',
      colorServerBg: 'rgb(214, 236, 255)',
      colorServerBgIsDark: false,
      colorServerFg: 'black',
      colorUiBg: 'white',
      colorUiBgIsDark: false,
      colorUiFg: 'rgb(64, 64, 64)',
      fCollapseAllNewStories: false,
      fDiscardRemoteClientLogs: false,
      fExpandAllNewAttachments: false,
      fShorthandForDuplicates: true,
      fShowClosedActions: false,
      forgetHysteresis: 0.25,
      maxRecords: 800,
      timeType: 'LOCAL',
    };
    const tree = renderer.create(
      <div>
        <Floats />
        <Settings
          onClose={() => {}}
          colors={BASE_COLORS}
          settings={settings}
          serverFilter="*:INFO"
          localClientFilter="*:DEBUG"
          updateSettings={() => {}}
          setServerFilter={() => {}}
          setLocalClientFilter={() => {}}
        />
      </div>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
