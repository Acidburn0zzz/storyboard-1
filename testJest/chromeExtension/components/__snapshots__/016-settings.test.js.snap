exports[`Settings renders correctly 1`] = `
<div>
  <div
    className="giu-floats"
    style={
      Object {
        "height": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 0,
      }
    } />
  <div
    className="giu-modal"
    id={undefined}
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "height": "100vh",
        "left": 0,
        "position": "fixed",
        "top": 0,
        "width": "100vw",
        "zIndex": 50,
      }
    }>
    <div
      className="giu-backdrop"
      onClick={undefined}
      onTouchMove={[Function]}
      onWheel={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "height": "100vh",
          "left": 0,
          "opacity": 0.7,
          "position": "fixed",
          "top": 0,
          "width": "100vw",
        }
      } />
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "left": 0,
          "position": "fixed",
          "top": "5vh",
          "width": "100%",
        }
      }>
      <div
        onClick={undefined}
        onTouchMove={[Function]}
        onWheel={[Function]}
        style={
          Object {
            "WebkitFlex": 1,
            "flex": 1,
          }
        } />
      <div
        onScroll={[Function]}
        onWheel={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 2,
            "boxShadow": "0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)",
            "color": "black",
            "maxHeight": "90vh",
            "overflowY": "auto",
            "padding": 20,
          }
        }>
        <input
          autoFocus={true}
          style={
            Object {
              "cursor": "default",
              "height": 0,
              "left": 8,
              "opacity": 0,
              "padding": 0,
              "pointerEvents": "none",
              "position": "fixed",
              "top": 8,
              "width": 0,
              "zIndex": -80,
            }
          }
          tabIndex={undefined} />
        <div
          className="allowUserSelect"
          style={
            Object {
              "border": "1px solid red",
              "borderRadius": 2,
              "color": "red",
              "marginBottom": 10,
              "padding": 15,
            }
          }>
          Changes to these settings can\'t be saved (beyond your current session) due to your current Chrome configuration. Please visit 
          <b>
            chrome://settings/content
          </b>
           and uncheck the option \"Block third-party cookies and site data\". Then close the Chrome DevTools and open them again.
        </div>
        <span
          className="giu-checkbox"
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }>
          <input
            checked={false}
            className={undefined}
            id="giu-checkbox_0"
            onBlur={[Function]}
            onChange={[Function]}
            onCopy={undefined}
            onCut={undefined}
            onFocus={[Function]}
            onPaste={undefined}
            required={undefined}
            tabIndex={undefined}
            type="checkbox" />
          <label
            htmlFor="giu-checkbox_0"
            style={undefined}>
            <span>
              Show 
              <i>
                CLOSED
              </i>
               actions
            </span>
          </label>
        </span>
        <br />
        <span
          className="giu-checkbox"
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }>
          <input
            checked={true}
            className={undefined}
            id="giu-checkbox_1"
            onBlur={[Function]}
            onChange={[Function]}
            onCopy={undefined}
            onCut={undefined}
            onFocus={[Function]}
            onPaste={undefined}
            required={undefined}
            tabIndex={undefined}
            type="checkbox" />
          <label
            htmlFor="giu-checkbox_1"
            style={undefined}>
            <span>
              Use shorthand notation for identical consecutive logs ( 
              <i
                className="giu-icon fa fa-copy"
                onClick={undefined}
                style={
                  Object {
                    "color": "currentColor",
                    "cursor": undefined,
                    "letterSpacing": "normal",
                  }
                } />
               )
            </span>
          </label>
        </span>
        <br />
        <span
          className="giu-checkbox"
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }>
          <input
            checked={false}
            className={undefined}
            id="giu-checkbox_2"
            onBlur={[Function]}
            onChange={[Function]}
            onCopy={undefined}
            onCut={undefined}
            onFocus={[Function]}
            onPaste={undefined}
            required={undefined}
            tabIndex={undefined}
            type="checkbox" />
          <label
            htmlFor="giu-checkbox_2"
            style={undefined}>
            Collapse all new stories (even if they are still open)
          </label>
        </span>
        <br />
        <span
          className="giu-checkbox"
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }>
          <input
            checked={false}
            className={undefined}
            id="giu-checkbox_3"
            onBlur={[Function]}
            onChange={[Function]}
            onCopy={undefined}
            onCut={undefined}
            onFocus={[Function]}
            onPaste={undefined}
            required={undefined}
            tabIndex={undefined}
            type="checkbox" />
          <label
            htmlFor="giu-checkbox_3"
            style={undefined}>
            Expand all attachments upon receipt
          </label>
        </span>
        <br />
        <span
          className="giu-checkbox"
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }>
          <input
            checked={false}
            className={undefined}
            id="giu-checkbox_4"
            onBlur={[Function]}
            onChange={[Function]}
            onCopy={undefined}
            onCut={undefined}
            onFocus={[Function]}
            onPaste={undefined}
            required={undefined}
            tabIndex={undefined}
            type="checkbox" />
          <label
            htmlFor="giu-checkbox_4"
            style={undefined}>
            Discard stories from remote clients upon receipt
          </label>
        </span>
        <br />
        <br />
        <div
          style={
            Object {
              "marginBottom": 5,
            }
          }>
          Log filters, e.g. 
          <b>
            foo, ba*:INFO, -test, *:WARN
          </b>
           
          <a
            href="https://github.com/guigrpa/storyboard#log-filtering"
            style={
              Object {
                "color": "currentColor",
              }
            }
            target="_blank">
            (more examples here)
          </a>
          :
        </div>
        <ul
          style={
            Object {
              "marginTop": 0,
            }
          }>
          <li>
            <label
              htmlFor="serverFilter"
              style={
                Object {
                  "display": "inline-block",
                  "width": 80,
                }
              }>
              Server:
            </label>
             
            <input
              className="giu-text-input"
              id="serverFilter"
              onBlur={[Function]}
              onChange={[Function]}
              onCopy={undefined}
              onCut={undefined}
              onFocus={[Function]}
              onPaste={undefined}
              orient={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "1px solid #999999",
                  "color": "inherit",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "width": 300,
                }
              }
              tabIndex={undefined}
              type="text"
              value="*:INFO" />
          </li>
          <li>
            <label
              htmlFor="localClientFilter"
              style={
                Object {
                  "display": "inline-block",
                  "width": 80,
                }
              }>
              Local client:
            </label>
             
            <input
              className="giu-text-input"
              id="localClientFilter"
              onBlur={[Function]}
              onChange={[Function]}
              onCopy={undefined}
              onCut={undefined}
              onFocus={[Function]}
              onPaste={undefined}
              orient={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "1px solid #999999",
                  "color": "inherit",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "width": 300,
                }
              }
              tabIndex={undefined}
              type="text"
              value="*:DEBUG" />
          </li>
        </ul>
        <div>
          <label
            htmlFor="maxRecords">
            Number of logs and stories to remember:
          </label>
           
          <input
            className="giu-number-input"
            id="maxRecords"
            min={0}
            onBlur={[Function]}
            onChange={[Function]}
            onCopy={undefined}
            onCut={undefined}
            onFocus={[Function]}
            onPaste={undefined}
            orient={undefined}
            step={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "1px solid #999999",
                "color": "inherit",
                "fontFamily": "inherit",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "width": 50,
              }
            }
            tabIndex={undefined}
            type="number"
            value="800" />
           
          <label
            htmlFor="forgetHysteresis">
            with hysteresis:
          </label>
           
          <input
            className="giu-number-input"
            id="forgetHysteresis"
            max={1}
            min={0}
            onBlur={[Function]}
            onChange={[Function]}
            onCopy={undefined}
            onCut={undefined}
            onFocus={[Function]}
            onPaste={undefined}
            orient={undefined}
            step={0.05}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "1px solid #999999",
                "color": "inherit",
                "fontFamily": "inherit",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "width": 50,
              }
            }
            tabIndex={undefined}
            type="number"
            value="0.25" />
           
          <i
            className="giu-icon fa fa-info-circle"
            style={
              Object {
                "color": undefined,
                "cursor": "pointer",
                "letterSpacing": "normal",
              }
            }
            title="When the backlog reaches 800, Storyboard will start forgetting old stuff until it goes below 600" />
        </div>
        <div>
          Colors: client stories:
           
          <span
            className="giu-color-input"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
                "top": 1,
              }
            }>
            <input
              onBlur={[Function]}
              onCopy={[Function]}
              onCut={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onPaste={[Function]}
              style={
                Object {
                  "cursor": "default",
                  "height": 0,
                  "left": 8,
                  "opacity": 0,
                  "padding": 0,
                  "pointerEvents": "none",
                  "position": "fixed",
                  "top": 8,
                  "width": 0,
                  "zIndex": -80,
                }
              }
              tabIndex={undefined} />
            <div
              onMouseDown={[Function]}
              style={
                Object {
                  "backgroundColor": "white",
                  "border": "1px solid #999999",
                  "color": "transparent",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 20,
                  "lineHeight": "1em",
                  "padding": "2px 4px 6px 4px",
                  "position": "relative",
                  "width": 35,
                }
              }>
              x
              <div
                className="giu-transparency-tiles"
                style={
                  Object {
                    "borderRadius": 2,
                    "bottom": 4,
                    "left": 4,
                    "position": "absolute",
                    "right": 4,
                    "top": 4,
                  }
                } />
              <div
                style={
                  Object {
                    "background": "rgb(240, 248, 255)",
                    "borderRadius": 2,
                    "bottom": 4,
                    "left": 4,
                    "position": "absolute",
                    "right": 4,
                    "top": 4,
                  }
                } />
            </div>
          </span>
           
          server stories:
           
          <span
            className="giu-color-input"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
                "top": 1,
              }
            }>
            <input
              onBlur={[Function]}
              onCopy={[Function]}
              onCut={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onPaste={[Function]}
              style={
                Object {
                  "cursor": "default",
                  "height": 0,
                  "left": 8,
                  "opacity": 0,
                  "padding": 0,
                  "pointerEvents": "none",
                  "position": "fixed",
                  "top": 8,
                  "width": 0,
                  "zIndex": -80,
                }
              }
              tabIndex={undefined} />
            <div
              onMouseDown={[Function]}
              style={
                Object {
                  "backgroundColor": "white",
                  "border": "1px solid #999999",
                  "color": "transparent",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 20,
                  "lineHeight": "1em",
                  "padding": "2px 4px 6px 4px",
                  "position": "relative",
                  "width": 35,
                }
              }>
              x
              <div
                className="giu-transparency-tiles"
                style={
                  Object {
                    "borderRadius": 2,
                    "bottom": 4,
                    "left": 4,
                    "position": "absolute",
                    "right": 4,
                    "top": 4,
                  }
                } />
              <div
                style={
                  Object {
                    "background": "rgb(214, 236, 255)",
                    "borderRadius": 2,
                    "bottom": 4,
                    "left": 4,
                    "position": "absolute",
                    "right": 4,
                    "top": 4,
                  }
                } />
            </div>
          </span>
           
          background:
           
          <span
            className="giu-color-input"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
                "top": 1,
              }
            }>
            <input
              onBlur={[Function]}
              onCopy={[Function]}
              onCut={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onPaste={[Function]}
              style={
                Object {
                  "cursor": "default",
                  "height": 0,
                  "left": 8,
                  "opacity": 0,
                  "padding": 0,
                  "pointerEvents": "none",
                  "position": "fixed",
                  "top": 8,
                  "width": 0,
                  "zIndex": -80,
                }
              }
              tabIndex={undefined} />
            <div
              onMouseDown={[Function]}
              style={
                Object {
                  "backgroundColor": "white",
                  "border": "1px solid #999999",
                  "color": "transparent",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "inherit",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 20,
                  "lineHeight": "1em",
                  "padding": "2px 4px 6px 4px",
                  "position": "relative",
                  "width": 35,
                }
              }>
              x
              <div
                className="giu-transparency-tiles"
                style={
                  Object {
                    "borderRadius": 2,
                    "bottom": 4,
                    "left": 4,
                    "position": "absolute",
                    "right": 4,
                    "top": 4,
                  }
                } />
              <div
                style={
                  Object {
                    "background": "rgb(255, 255, 255)",
                    "borderRadius": 2,
                    "bottom": 4,
                    "left": 4,
                    "position": "absolute",
                    "right": 4,
                    "top": 4,
                  }
                } />
            </div>
          </span>
          <div
            style={
              Object {
                "marginTop": 3,
              }
            }>
            (Use very light or very dark colors for best results, or choose a
             
            <span
              className="giu-button"
              onClick={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "border": "1px solid #999999",
                  "borderRadius": 5,
                  "cursor": "pointer",
                  "display": "inline-block",
                  "padding": "1px 5px",
                  "userSelect": "none",
                }
              }>
              preset
            </span>
            )
          </div>
        </div>
        <div
          style={
            Object {
              "borderTop": "1px solid #999999",
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 10,
              "paddingTop": 10,
            }
          }>
          <span
            className="giu-button"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "border": "1px solid #999999",
                "borderRadius": 5,
                "cursor": "pointer",
                "display": "inline-block",
                "marginRight": 5,
                "padding": "1px 5px",
                "userSelect": "none",
              }
            }>
            Cancel
          </span>
          <span
            className="giu-button"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "border": "1px solid #999999",
                "borderRadius": 5,
                "cursor": "pointer",
                "display": "inline-block",
                "marginRight": 5,
                "padding": "1px 5px",
                "userSelect": "none",
              }
            }>
            Reset defaults
          </span>
          <div
            style={
              Object {
                "WebkitFlex": 1,
                "flex": 1,
              }
            } />
          <span
            className="giu-button"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "border": "1px solid black",
                "borderRadius": 5,
                "cursor": "pointer",
                "display": "inline-block",
                "marginLeft": 5,
                "padding": "1px 5px",
                "userSelect": "none",
              }
            }>
            Save
          </span>
        </div>
      </div>
      <div
        onClick={undefined}
        onTouchMove={[Function]}
        onWheel={[Function]}
        style={
          Object {
            "WebkitFlex": 1,
            "flex": 1,
          }
        } />
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
  <div>
    <div
      class="giu-floats"
      style="height:0;left:0;position:absolute;top:0;width:0;">
    </div>
    <div
      class="giu-modal"
      style="height:100vh;left:0;position:fixed;top:0;width:100vw;z-index:50;">
      <div
        class="giu-backdrop"
        style="background-color:white;height:100vh;left:0;opacity:0.7;position:fixed;top:0;width:100vw;">
      </div>
      <div
        style="display:flex;flex-direction:row;left:0;position:fixed;top:5vh;width:100%;">
        <div
          style="-webkit-flex:1;flex:1;">
        </div>
        <div
          style="background-color:white;border-radius:2px;box-shadow:0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);color:black;max-height:90vh;overflow-y:auto;padding:20px;">
          <input
            autoFocus
            style="cursor:default;height:0;left:8px;opacity:0;padding:0;pointer-events:none;position:fixed;top:8px;width:0;z-index:-80;">
          </input>
          <div
            class="allowUserSelect"
            style="border:1px solid red;border-radius:2px;color:red;margin-bottom:10px;padding:15px;">
            Changes to these settings can\'t be saved (beyond your current session) due to your current Chrome configuration. Please visit 
            <b>
              chrome://settings/content
            </b>
             and uncheck the option \"Block third-party cookies and site data\". Then close the Chrome DevTools and open them again.
          </div>
          <span
            class="giu-checkbox"
            style="white-space:nowrap;">
            <input
              id="giu-checkbox_0"
              type="checkbox">
            </input>
            <label
              for="giu-checkbox_0">
              <span>
                Show 
                <i>
                  CLOSED
                </i>
                 actions
              </span>
            </label>
          </span>
          <br>
          </br>
          <span
            class="giu-checkbox"
            style="white-space:nowrap;">
            <input
              checked
              id="giu-checkbox_1"
              type="checkbox">
            </input>
            <label
              for="giu-checkbox_1">
              <span>
                Use shorthand notation for identical consecutive logs ( 
                <i
                  class="giu-icon fa fa-copy"
                  style="color:currentColor;letter-spacing:normal;">
                </i>
                 )
              </span>
            </label>
          </span>
          <br>
          </br>
          <span
            class="giu-checkbox"
            style="white-space:nowrap;">
            <input
              id="giu-checkbox_2"
              type="checkbox">
            </input>
            <label
              for="giu-checkbox_2">
              Collapse all new stories (even if they are still open)
            </label>
          </span>
          <br>
          </br>
          <span
            class="giu-checkbox"
            style="white-space:nowrap;">
            <input
              id="giu-checkbox_3"
              type="checkbox">
            </input>
            <label
              for="giu-checkbox_3">
              Expand all attachments upon receipt
            </label>
          </span>
          <br>
          </br>
          <span
            class="giu-checkbox"
            style="white-space:nowrap;">
            <input
              id="giu-checkbox_4"
              type="checkbox">
            </input>
            <label
              for="giu-checkbox_4">
              Discard stories from remote clients upon receipt
            </label>
          </span>
          <br>
          </br>
          <br>
          </br>
          <div
            style="margin-bottom:5px;">
            Log filters, e.g. 
            <b>
              foo, ba*:INFO, -test, *:WARN
            </b>
             
            <a
              href="https://github.com/guigrpa/storyboard#log-filtering"
              style="color:currentColor;"
              target="_blank">
              (more examples here)
            </a>
            :
          </div>
          <ul
            style="margin-top:0;">
            <li>
              <label
                for="serverFilter"
                style="display:inline-block;width:80px;">
                Server:
              </label>
               
              <input
                class="giu-text-input"
                id="serverFilter"
                style="background-color:transparent;border:1px solid #999999;color:inherit;font-family:inherit;font-size:inherit;font-weight:inherit;width:300px;"
                type="text"
                value="*:INFO">
              </input>
            </li>
            <li>
              <label
                for="localClientFilter"
                style="display:inline-block;width:80px;">
                Local client:
              </label>
               
              <input
                class="giu-text-input"
                id="localClientFilter"
                style="background-color:transparent;border:1px solid #999999;color:inherit;font-family:inherit;font-size:inherit;font-weight:inherit;width:300px;"
                type="text"
                value="*:DEBUG">
              </input>
            </li>
          </ul>
          <div>
            <label
              for="maxRecords">
              Number of logs and stories to remember:
            </label>
             
            <input
              class="giu-number-input"
              id="maxRecords"
              style="background-color:transparent;border:1px solid #999999;color:inherit;font-family:inherit;font-size:inherit;font-weight:inherit;width:50px;"
              type="number"
              value="800">
            </input>
             
            <label
              for="forgetHysteresis">
              with hysteresis:
            </label>
             
            <input
              class="giu-number-input"
              id="forgetHysteresis"
              style="background-color:transparent;border:1px solid #999999;color:inherit;font-family:inherit;font-size:inherit;font-weight:inherit;width:50px;"
              type="number"
              value="0.25">
            </input>
             
            <i
              class="giu-icon fa fa-info-circle"
              style="cursor:pointer;letter-spacing:normal;"
              title="When the backlog reaches 800, Storyboard will start forgetting old stuff until it goes below 600">
            </i>
          </div>
          <div>
            Colors: client stories:
             
            <span
              class="giu-color-input"
              style="display:inline-block;position:relative;top:1px;">
              <input
                style="cursor:default;height:0;left:8px;opacity:0;padding:0;pointer-events:none;position:fixed;top:8px;width:0;z-index:-80;">
              </input>
              <div
                style="background-color:white;border:1px solid #999999;color:transparent;cursor:pointer;display:inline-block;font-family:inherit;font-size:inherit;font-weight:inherit;height:20px;line-height:1em;padding:2px 4px 6px 4px;position:relative;width:35px;">
                x
                <div
                  class="giu-transparency-tiles"
                  style="border-radius:2px;bottom:4px;left:4px;position:absolute;right:4px;top:4px;">
                </div>
                <div
                  style="background:rgb(240, 248, 255);border-radius:2px;bottom:4px;left:4px;position:absolute;right:4px;top:4px;">
                </div>
              </div>
            </span>
             
            server stories:
             
            <span
              class="giu-color-input"
              style="display:inline-block;position:relative;top:1px;">
              <input
                style="cursor:default;height:0;left:8px;opacity:0;padding:0;pointer-events:none;position:fixed;top:8px;width:0;z-index:-80;">
              </input>
              <div
                style="background-color:white;border:1px solid #999999;color:transparent;cursor:pointer;display:inline-block;font-family:inherit;font-size:inherit;font-weight:inherit;height:20px;line-height:1em;padding:2px 4px 6px 4px;position:relative;width:35px;">
                x
                <div
                  class="giu-transparency-tiles"
                  style="border-radius:2px;bottom:4px;left:4px;position:absolute;right:4px;top:4px;">
                </div>
                <div
                  style="background:rgb(214, 236, 255);border-radius:2px;bottom:4px;left:4px;position:absolute;right:4px;top:4px;">
                </div>
              </div>
            </span>
             
            background:
             
            <span
              class="giu-color-input"
              style="display:inline-block;position:relative;top:1px;">
              <input
                style="cursor:default;height:0;left:8px;opacity:0;padding:0;pointer-events:none;position:fixed;top:8px;width:0;z-index:-80;">
              </input>
              <div
                style="background-color:white;border:1px solid #999999;color:transparent;cursor:pointer;display:inline-block;font-family:inherit;font-size:inherit;font-weight:inherit;height:20px;line-height:1em;padding:2px 4px 6px 4px;position:relative;width:35px;">
                x
                <div
                  class="giu-transparency-tiles"
                  style="border-radius:2px;bottom:4px;left:4px;position:absolute;right:4px;top:4px;">
                </div>
                <div
                  style="background:rgb(255, 255, 255);border-radius:2px;bottom:4px;left:4px;position:absolute;right:4px;top:4px;">
                </div>
              </div>
            </span>
            <div
              style="margin-top:3px;">
              (Use very light or very dark colors for best results, or choose a
               
              <span
                class="giu-button"
                style="-webkit-user-select:none;border:1px solid #999999;border-radius:5px;cursor:pointer;display:inline-block;padding:1px 5px;user-select:none;">
                preset
              </span>
              )
            </div>
          </div>
          <div
            style="border-top:1px solid #999999;display:flex;flex-direction:row;margin-top:10px;padding-top:10px;">
            <span
              class="giu-button"
              style="-webkit-user-select:none;border:1px solid #999999;border-radius:5px;cursor:pointer;display:inline-block;margin-right:5px;padding:1px 5px;user-select:none;">
              Cancel
            </span>
            <span
              class="giu-button"
              style="-webkit-user-select:none;border:1px solid #999999;border-radius:5px;cursor:pointer;display:inline-block;margin-right:5px;padding:1px 5px;user-select:none;">
              Reset defaults
            </span>
            <div
              style="-webkit-flex:1;flex:1;">
            </div>
            <span
              class="giu-button"
              style="-webkit-user-select:none;border:1px solid black;border-radius:5px;cursor:pointer;display:inline-block;margin-left:5px;padding:1px 5px;user-select:none;">
              Save
            </span>
          </div>
        </div>
        <div
          style="-webkit-flex:1;flex:1;">
        </div>
      </div>
    </div>
  </div>
`;
